src/main/java/models/ProductModel.java

```diff
@@ -0,0 +1,21 @@
+package models;
+
+import enums.Currency;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+public class ProductModel {
+	public final String name;
+	public final Currency priceCurrency;
+	public final Double price;
```

Comment:
Java `BigDecimal` would be the best to store prices - same thought for `priceBeforeSale`.
Float and double are prone to losing the precision.
https://medium.com/@cancerian0684/which-data-type-would-you-choose-for-storing-currency-values-like-trading-price-dd7489e7a439

Reply:
Another interesting topic ❤️  Type of variables changed to `BigDecimal`, thanks